sources/packages/api/src/route-controllers/education-program/education-program.controller.service.ts

```diff
@@ -0,0 +1,205 @@
+      if (error instanceof CustomNamedError) {
+        if (error.name === EDUCATION_PROGRAM_NOT_FOUND) {
+          throw new NotFoundException(error.message);
+        }
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * Education program information shared between the Ministry and the Institution.
+   * @param programId program id.
+   * @param institutionId when provided, ensures the proper authorization
+   * checking if the institution has access to the program.
+   * @returns programs information.
+   * */
+  async getEducationProgram(
+    programId: number,
+    institutionId?: number,
+  ): Promise<EducationProgramAPIOutDTO> {
+    const programPromise = this.programService.getEducationProgramDetails(
+      programId,
+      institutionId,
+    );
+    const hasOfferingsPromise =
+      this.educationProgramOfferingService.hasExistingOffering(programId);
+    // Wait for the program and for the offering check to be retrieved.
+    const [program, hasOfferings] = await Promise.all([
+      programPromise,
+      hasOfferingsPromise,
+    ]);
+
+    if (!program) {
+      throw new NotFoundException("Not able to find the requested program.");
+    }
+
+    return {
+      id: program.id,
+      programStatus: program.programStatus,
+      name: program.name,
+      description: program.description,
+      credentialType: program.credentialType,
+      credentialTypeToDisplay: credentialTypeToDisplay(program.credentialType),
+      cipCode: program.cipCode,
+      nocCode: program.nocCode,
+      sabcCode: program.sabcCode,
+      regulatoryBody: program.regulatoryBody,
+      programDeliveryTypes: {
+        deliveredOnSite: program.deliveredOnSite,
+        deliveredOnline: program.deliveredOnline,
+      },
+      deliveredOnlineAlsoOnsite: program.deliveredOnlineAlsoOnsite,
+      sameOnlineCreditsEarned: program.sameOnlineCreditsEarned,
+      earnAcademicCreditsOtherInstitution:
+        program.earnAcademicCreditsOtherInstitution,
+      courseLoadCalculation: program.courseLoadCalculation,
+      completionYears: program.completionYears,
+      eslEligibility: program.eslEligibility,
+      hasJointInstitution: program.hasJointInstitution,
+      hasJointDesignatedInstitution: program.hasJointDesignatedInstitution,
+      programIntensity: program.programIntensity,
+      institutionProgramCode: program.institutionProgramCode,
+      minHoursWeek: program.minHoursWeek,
+      isAviationProgram: program.isAviationProgram,
+      minHoursWeekAvi: program.minHoursWeekAvi,
+      entranceRequirements: {
+        hasMinimumAge: program.hasMinimumAge,
+        minHighSchool: program.minHighSchool,
+        requirementsByInstitution: program.requirementsByInstitution,
+        requirementsByBCITA: program.requirementsByBCITA,
+      },
+      hasWILComponent: program.hasWILComponent,
+      isWILApproved: program.isWILApproved,
+      wilProgramEligibility: program.wilProgramEligibility,
+      hasTravel: program.hasTravel,
```

Comment:
Please what does hasTravel mean ? Is this supposed to be like travel allowance ?

Reply:
I do not have much information about it. The comments say:
"Identifier for Field trip, field placement, or travel component in program."
It was introduced 8 months ago.